src/Storages/StorageURL.cpp

```diff
@@ -967,6 +967,16 @@ URLBasedDataSourceConfiguration StorageURL::getConfiguration(ASTs & args, Contex
             configuration.compression_method = checkAndGetLiteralArgument<String>(args[2], "compression_method");
     }
 
+    /// Make sure compressionMethod is correct
+    try
+    {
+        chooseCompressionMethod(configuration.url, configuration.compression_method);
```

Comment:
There is `String compression_method` in `StorageURL` class. Could you please change it to `CompressionMethod compression_method` and initialise with `chooseCompressionMethod(configuration.url, configuration.compression_method);` in constructor. It would be better and will also throw exception and prevent storage creation.

Reply:
Updated